app/js/components/login/Login.jsx

```diff
@@ -53,9 +68,16 @@ class Login extends Component {
   )
 
   handleResponse = (response) => {
-    response.data.authenticated
-      ? this.props.history.push('/')
-      : this.toastError('Incorrect username or password');
+    const { formSubmitted } = this.state;
+    this.setState({
+      loaded: true,
+      formSubmitted: false,
+    });
+    if (response.data.authenticated) {
+      this.props.history.push('/');
+    } else if (response.data.authenticated === false && formSubmitted) {
+      this.toastError('Incorrect username or password');
+    }
```

Comment:
this should be done in your actions and reducers as well.

Reply:
This would be appropriate if it was a redux application. Thanks, @Efosaok, for the reviews.